privacyidea/lib/tokenclass.py

```diff
@@ -190,9 +190,9 @@ def set_user(self, user, report=None):
         :return: None
         """
         (uid, resolvertype, resolvername) = user.get_user_identifiers()
-        self.token.resolver = resolvername
-        self.token.resolver_type = resolvertype
-        self.token.user_id = uid
+        r = TokenOwner(token_id=self.token.id,
+                       user_id=uid, resolver=resolvername,
+                       resolver_type=resolvertype, realmname=user.realm).save()
```

Comment:
Hm, what happens if the token already has a user? In the old behavior, the user was just overwritten. I think here, the token would be assigned a second user?